src/components/BatchClassComponents/AlertDialog.js

```diff
@@ -0,0 +1,164 @@
+  const {
+    open,
+    close,
+    title,
+    start_time,
+    end_time,
+    id,
+    registerAll,
+    exerciseReload,
+    setIsEnrolled,
+    type,
+  } = props;
+  console.log("props", props);
+  const user = useSelector(({ User }) => User);
+  const params = useParams();
+  const pathwayId = params.pathwayId;
+  const dispatch = useDispatch();
+  const [loading, setLoading] = React.useState(false);
+  const handelEnrollment = (Id) => {
+    setLoading(true);
+    axios
+      .post(
+        `${
+          process.env.REACT_APP_MERAKI_URL
+        }/classes/${Id}/register?register-all=${registerAll || false}`,
+        {},
+        {
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: user.data.token,
+          },
+        }
+      )
+      .then(() => {
+        toast.success("Class Enrolled", {
+          position: toast.POSITION.BOTTOM_RIGHT,
+          autoClose: 2500,
+        });
+        setLoading(false);
+
+        close();
+        if (registerAll) {
+          dispatch(
+            enrolledBatchesActions.getEnrolledBatches({
+              pathwayId: pathwayId,
+              authToken: user?.data?.token,
+            })
+          );
+          dispatch(
+            upcomingClassActions.getupcomingEnrolledClasses({
+              pathwayId: pathwayId,
+              authToken: user?.data?.token,
+            })
+          );
+        } else if (setIsEnrolled) {
+          setIsEnrolled(true);
+        }
+      })
+      .catch((err) => {
+        setLoading(false);
+        toast.error("Failed To Enroll To Class", {
+          position: toast.POSITION.BOTTOM_RIGHT,
+          autoClose: 2500,
+        });
+        close();
+      });
+  };
+  return (
+    <div>
+      <Dialog open={open} onClose={close}>
+        <DialogContent sx={{ maxWidth: 370 }}>
+          {type == "batch" ? (
+            <>
+              <Typography variant="h6" sx={{ fontFamily: "Lusitana" }}>
+                Awesome! You have taken the first step to being a programmer
+              </Typography>
+              <Typography variant="h6" mt={2} sx={{ fontFamily: "Lusitana" }}>
+                Batch:{title}
+              </Typography>
+            </>
+          ) : type == "DoubtClass" ? (
+            <>
+              <Typography variant="h6" sx={{ fontFamily: "Lusitana" }}>
+                Doubt Class
+              </Typography>
+              <Typography variant="h6" mt={2} sx={{ fontFamily: "Lusitana" }}>
+                Class : {title}
+              </Typography>
+            </>
+          ) : type == "RevisionClass" ? (
+            <>
+              <Typography variant="h6" sx={{ fontFamily: "Lusitana" }}>
+                Revision Class
+              </Typography>
+              <Typography variant="h6" mt={2} sx={{ fontFamily: "Lusitana" }}>
+                From:{title}
+              </Typography>
+            </>
+          ) : (
+            ""
+          )}
```

Comment:
You can add a map from types of classes to Arrays of text like this:
```
const dialogContentMap = {
  batch: ['Awesome! You have taken the first step to being a programmer', 'Batch: ' + title],
  DoubtClass: ['Doubt Class', 'Class : ' + title],
  RevisionClass: ['Revision Class', 'From: ' + title]
};
```
And then use the following here:
```suggestion
          {dialogContentMap[type]?.map((line, num) => (
            <Typography variant="h6" mt={num && 2} sx={{ fontFamily: 'Lusitana' }}>
              {line}
            </Typography>
          ))}
```